src/test/regress/sql/seclabel_non_maindb.sql

```diff
@@ -0,0 +1,72 @@
+-- SECLABEL
+--
+-- Test suite for running SECURITY LABEL ON ROLE statements from non-main databases
+
+SET citus.enable_create_database_propagation to ON;
+
+CREATE DATABASE database1;
+CREATE DATABASE database2;
+
+\c - - - :worker_1_port
+SET citus.enable_create_database_propagation to ON;
+CREATE DATABASE database_w1;
+
+
+\c - - - :master_port
+CREATE ROLE user1;
+\c database1
+SHOW citus.main_db;
+SHOW citus.superuser;
+
+CREATE ROLE "user 2";
+
+-- Set a SECURITY LABEL on a role from a non-main database
+SECURITY LABEL FOR "citus '!tests_label_provider" ON ROLE user1 IS 'citus_classified';
+SECURITY LABEL FOR "citus '!tests_label_provider" ON ROLE "user 2" IS 'citus_unclassified';
+
+-- Check the result
+\c regression
+SELECT node_type, result FROM get_citus_tests_label_provider_labels('user1') ORDER BY node_type;
+SELECT node_type, result FROM get_citus_tests_label_provider_labels('"user 2"') ORDER BY node_type;
+
+\c database1
+-- Set a SECURITY LABEL on database, it should not be propagated
+SECURITY LABEL FOR "citus '!tests_label_provider" ON DATABASE database1 IS 'citus_classified';
+
+-- Set a SECURITY LABEL on a table, it should not be propagated
+CREATE TABLE a (i int);
+SECURITY LABEL ON TABLE a IS 'citus_classified';
+
+\c regression
+SELECT node_type, result FROM get_citus_tests_label_provider_labels('database1') ORDER BY node_type;
+
+-- Check that only the SECURITY LABEL for ROLES is propagated to the non-main databases on other nodes
+\c - - - :worker_1_port
+\c database_w1
+SELECT provider, objtype, label, objname FROM pg_seclabels ORDER BY objname;
+
```

Comment:
We already ensure this a few lines ago via the following:
```sql
-- Check the result
\c regression
SELECT node_type, result FROM get_citus_tests_label_provider_labels('user1') ORDER BY node_type;
SELECT node_type, result FROM get_citus_tests_label_provider_labels('"user 2"') ORDER BY node_type;
```

So can remove those lines:
```suggestion
```

Reply:
database_w1 does not have Citus extension, whereas regression database has.  In addition, at line 46 we retrieve all objects with security labels. That check ensures that we only propagate security labels associated with roles.

Reply:
I got it now 👍

And besides that, we can combine those two lines:
```sql
\c - - - :worker_1_port
\c database_w1
```

into

```sql
\c database_w1 - - :worker_1_port
```

to switch both the database and the server port at once.

Reply:
Neat, updated it